File: app/src/main/java/org/neidhardt/dynamicsoundboard/dialog/soundlayouts/AddNewSoundLayoutDialog.java
package org.neidhardt.dynamicsoundboard.dialog.soundlayouts;

import android.app.FragmentManager;
import android.os.Bundle;
import org.neidhardt.dynamicsoundboard.R;
import org.neidhardt.dynamicsoundboard.dao.SoundLayout;
import org.neidhardt.dynamicsoundboard.navigationdrawer.NavigationDrawerFragment;
import org.neidhardt.dynamicsoundboard.soundlayouts.SoundLayoutsManager;

/**
 * Created by eric.neidhardt on 12.03.2015.
 */
public class AddNewSoundLayoutDialog extends SoundLayoutDialog
{
	private static final String TAG = AddNewSoundLayoutDialog.class.getName();

	private static final String KEY_SUGGESTED_NAME = "org.neidhardt.dynamicsoundboard.dialog.AddNewSoundLayoutDialog.suggestedName";

	private String suggestedName;

	public static void showInstance(FragmentManager manager, String suggestedName)
	{
		AddNewSoundLayoutDialog dialog = new AddNewSoundLayoutDialog();

		Bundle args = new Bundle();
		args.putString(KEY_SUGGESTED_NAME, suggestedName);
		dialog.setArguments(args);

		dialog.show(manager, TAG);
	}

	@Override
	public void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);

		Bundle args = this.getArguments();
		if (args != null)
			this.suggestedName = args.getString(KEY_SUGGESTED_NAME);
	}

	@Override
	protected int getLayoutId()
	{
		return R.layout.dialog_add_new_sound_layout;
	}

	@Override
	protected String getHintForName()
	{
		return this.suggestedName;
	}

	@Override
	protected void deliverResult()
	{
		String name = super.soundLayoutName.getDisplayedText();
		SoundLayout layout = new SoundLayout();
		layout.setIsSelected(false);
		layout.setDatabaseId(SoundLayoutsManager.getNewDatabaseIdForLabel(name));
		layout.setLabel(name);

		SoundLayoutsManager.getInstance().addSoundLayout(layout);

		NavigationDrawerFragment fragment = this.getNavigationDrawerFragment();
		if (fragment != null)
			fragment.triggerSoundLayoutUpdate();
	}
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/soundlayoutmanagement/views/AddNewSoundLayoutDialog.kt
package org.neidhardt.dynamicsoundboard.soundlayoutmanagement.views

import android.os.Bundle
import android.support.v4.app.FragmentManager
import org.neidhardt.dynamicsoundboard.R
import org.neidhardt.dynamicsoundboard.SoundboardApplication
import org.neidhardt.dynamicsoundboard.manager.NewSoundLayoutManager
import org.neidhardt.dynamicsoundboard.persistance.model.NewSoundLayout
import rx.subscriptions.CompositeSubscription

/**
 * File created by eric.neidhardt on 12.03.2015.
 */
class AddNewSoundLayoutDialog : SoundLayoutDialog() {

	private var suggestedName: String? = null

	private val manager = SoundboardApplication.newSoundLayoutManager

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)

		val args = this.arguments
		if (args != null)
			this.suggestedName = this.getHintForName()
	}

	override fun deliverResult() {
		var name = super.soundLayoutName?.text.toString()
		if (name.isEmpty())
			name = this.getHintForName()

		val layout = NewSoundLayout().apply {
			this.isSelected = false
			this.databaseId = NewSoundLayoutManager.getNewDatabaseIdForLabel(name)
			this.label = name
		}
		this.manager.add(layout)
		this.dismiss()
	}

	override fun getLayoutId(): Int = R.layout.dialog_add_new_sound_layout

	override fun getHintForName(): String = this.manager.getSuggestedName()

    override fun getTitleId(): Int = R.string.dialog_add_new_sound_layout_title

	override fun getPositiveButtonId(): Int = R.string.dialog_add

	companion object
	{
		private val TAG = AddNewSoundLayoutDialog::class.java.name

		fun showInstance(manager: FragmentManager) {
			val dialog = AddNewSoundLayoutDialog()
			dialog.show(manager, TAG)
		}
	}
}
